contracts/test/recovery-test.ts

```diff
@@ -54,7 +52,12 @@ describe("Recovery", async function () {
   });
 
   it("should update bls key", async function () {
-    const newKey: PublicKey = [1, 2, 3, 4].map(BigNumber.from);
+    const newKey: PublicKey = [
+      BigNumber.from(1),
+      BigNumber.from(2),
+      BigNumber.from(3),
+      BigNumber.from(4),
+    ];
```

Comment:
New Typescript version caught this. Is there a better way we could to this?

Reply:
> New Typescript version caught this.

I don't think that's what happened :/. The issue is that the type of `.map` works on arrays, not tuples, so using `.map` converts your tuple into an array, so the assignment absolutely shouldn't work. This hasn't changed, I checked back to 3.3.3 and it still 'correctly' rejects constructions like this:

https://www.typescriptlang.org/play?ts=3.3.3#code/C4TwDgpgBACg0lAvFA2gOwK4FsBGEBOANFJrgcaXkSdlQLoDcAUEwGYZoDGwAlgPZoowPgDlaBABQBnYPgBcUGfh5oA5gEoFlAlADeTKIaj4IwDPkFiy+abPXMAvi04CZUMAGsF8JKgDkAIx+xH4ATMFQfgDMEX4ALH50AHRYAIZgEsJWVPZMQA

I'm curious how this happened. Any ideas?

...the presence of CI would have been useful here :(

Anyway, in terms of a better way, I don't think you can. I mean, I think it's possible to write a `map` function that is tuple-aware, but I don't think it's worth doing here.

Reply:
Right, we could do `...map<PublicKey>(...)` or something similar, but better to be explicit.

I attempted to update your example to more closely match a `BigNumber` [example](https://www.typescriptlang.org/play?ts=3.3.3#code/C4TwDgpgBAQglgcwHIFcC2AjCAnOBnACygF5ZFVMcoAfKDROAO2Bqj2F0YVcfS2wBQoSFAAKAaRJQA2vGR8c+AgBoy8yrkKq5Ffku3kFmggF0A3AIEBjADYBDPHjW6qAbwFRPUKwHtG7bBQrYB9sAAoPLyiwFAwbOCsobAg7ABM-GxAoADc7GxQIAC4oXg1IzwBKKFcAX0to2PjE9jtgBKgAM2wfNDDcm2K7RhAK4p0javKoqBsIFgxGADU84tL+KSQ7JAtprzgOqDDhCB8D-pJiUgByAKYEK6r3Xd2F5ZspMDtsPAgASWY+nkKjtnjUoBAbD8oPtDsdTjk8hdrmscA9Js9pq9EaR+iDdnUplFksAUNhGCUIAB3ZxGMJYmzAqZ1Am+fwsMAAa2KEik0iuAEYrqorgAmIVQK4AZnFVwALFcTAA6NB2MBhcYaRVdHqMoA) but still get the same error. Unsure what changed to now get that error.

+1 Getting some CI processes in place, can prioritize after current round of work.